HP-1001: Fix error handling in profile api fetch

NikoHelle opened the pull request:
Error handling was incorrect when api returned any other error than graphQL error

API does not return an error when a profile does not exist, just an empty myProfile.

So if an error is returned, it is always fatal whether it is a graphQL or a network error.

Added better notfication for the user and buttons for retry and logout.

NikoHelle committed:
(HP-1001) Fix error handling in profile api fetch

Error handling was incorrect when api returned any other error than graphQL error

APi does not return an error when a profile does not exist, just an empty myProfile.

So if an error is returned, it is always fatal whether it is a graphQL or a network error

akikoskinen commented:
Some discoveries. Not all of these are necessarily part of this PR.

- The header bar dropdown menus (e.g. language selection) go behind the error notification.
- When the new error notification is shown, page title still says "Create profile".
- The buttons below the error notification have no spacing to the footer. Perhaps this gets fixed with #199 or some other PR?
- The error text could be a tad shorter. Perhaps it includes a bit more details than is necessary.

NikoHelle commented:
I rebased to develop -branch, which has never version of HDS. Now the Notification is not above menus
Also the error text has been cut and now the page title has error title.
The new footer and updated HDS did not improve padding, so added more.

akikoskinen reviewed (approved):
Looks good.
